rbac/management/management/commands/utils.py

```diff
@@ -132,3 +134,77 @@ def populate_service_account_data(file_name):
         if user_id := id_mapping.get(principal.service_account_id):
             principal.user_id = user_id
             principal.save()
+
+
+def populate_workspace_data(file_name, batch_size=250):
+    """Populate workspace data from the downloaded file."""
+    file_path = get_file_path(file_name)
+    current_line = 0
+    with open(file_path, "r") as file:
+        csv_reader = csv.DictReader(file)
+        batch_data = []
+        for row in csv_reader:
+            current_line += 1
+            batch_data.append(row)
+            if len(batch_data) >= batch_size:
+                batch_import_workspace(batch_data)
+                logger.info(f"Processed batch ending at line {current_line}")
+                batch_data = []
+        # Process any remaining records
+        if batch_data:
+            batch_import_workspace(batch_data)
+            logger.info(f"Processed final batch ending at line {current_line}")
+    return
+
+
+def batch_import_workspace(records):
+    """Import workspace records in batch."""
+    with transaction.atomic():
```

Comment:
@astrozzc is this ok or should we put everything in one transaction ? if it will fails I guess  `batch import workspace`method  is idempotent ?

Reply:
If we put everything in one transaction, the performance would be worse because the data is too large. Batched transaction should be enough.
Right, if we run again, it is supposed to be idempotent